notebooks/02_model/mmlspark_lightgbm_criteo.ipynb

```diff
@@ -0,0 +1,407 @@
+   "source": [
+    "# Content Based Personalization\n",
+    "## LightGBM on Azure Databricks<br>\n",
+    "This notebook provides a quick example of how to train LightGBM model on Azure Databricks and deploy it using MML Spark for a content personalization scenario.<br><br>\n",
+    "LightGBM \\[1\\] is a gradient boosting framework that uses tree-based learning algorithms.<br>\n",
+    "MML Spark \\[2\\] allows LightGBM to be called in a Spark environment which provides several advantages:\n",
+    "- Distributed computation for model development\n",
+    "- Easy integration into existing Spark workflows\n",
+    "- Model serving through Spark Serving \\[3\\]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Global Settings and Imports"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A python script is provided to simplify setting up Azure Databricks with the correct\n",
+    "dependencies.<br> Run ```python scripts/databricks_install.py -h``` for more details."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "System version: 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) \n",
+      "[GCC 7.3.0]\n",
+      "PySpark version: 2.3.1\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "from tempfile import TemporaryDirectory\n",
+    "sys.path.append(\"../../\")\n",
+    "\n",
+    "import pyspark\n",
+    "from pyspark.ml import PipelineModel\n",
+    "from pyspark.ml.feature import FeatureHasher\n",
+    "from pyspark.sql.functions import col, udf\n",
+    "from pyspark.sql.types import FloatType\n",
+    "import requests\n",
+    "\n",
+    "from reco_utils.common.spark_utils import start_or_get_spark\n",
+    "from reco_utils.common.notebook_utils import is_databricks\n",
+    "from reco_utils.dataset.criteo import load_spark_df\n",
+    "from reco_utils.dataset.spark_splitters import spark_random_split\n",
+    "\n",
+    "print(\"System version: {}\".format(sys.version))\n",
+    "print(\"PySpark version: {}\".format(pyspark.version.__version__))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Setup MML Spark\n",
+    "if not is_databricks():\n",
+    "    # get the maven coordinates for MML Spark from databricks_install script\n",
+    "    from scripts.databricks_install import MMLSPARK_INFO\n",
+    "    packages = [MMLSPARK_INFO['maven']['coordinates']]\n",
+    "    spark = start_or_get_spark(packages=packages)\n",
+    "\n",
+    "from mmlspark import ComputeModelStatistics\n",
+    "from mmlspark import LightGBMClassifier"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "ModuleNotFoundError",
```

Comment:
Are we going to use `azureml` in the notebook?

Reply:
that's a good question, if we use PipelineModel to save the model:

```
tmp = TemporaryDirectory()

# save model to temporary directory
model_name = 'finished.model'
save_file = os.path.join(tmp.name, model_name)
pipeline = PipelineModel(stages=[feature_processor, model])
pipeline.save(save_file)
```

we can use the o16n notebook to have this code:
```
# zip file for transfer to Azure ML Workspace
zip_file = shutil.make_archive(base_name=model_name, 
                               format='zip', 
                               root_dir='/dbfs{}'.format(save_file))

# register model in Azure ML Workspace
ws = Workspace.from_config()
model = Model.register(workspace=ws, model_path=zip_file, model_name=model_name)
```
this is possible right?

Reply:
this is fixed